actioncable/app/javascript/action_cable/index.js

```diff
@@ -0,0 +1,61 @@
+import Connection from "./connection"
+import ConnectionMonitor from "./connection_monitor"
+import Consumer from "./consumer"
+import INTERNAL from "./internal"
+import Subscription from "./subscription"
+import Subscriptions from "./subscriptions"
+
+const ActionCable = {
+  WebSocket: window.WebSocket,
+  logger: window.console,
+
+  createConsumer(url) {
+    if (url == null) {
+      const urlConfig = this.getConfig("url")
+      url = (urlConfig ? urlConfig : this.INTERNAL.default_mount_path)
+    }
+    return new ActionCable.Consumer(this.createWebSocketURL(url))
+  },
+
+  getConfig(name) {
+    const element = document.head.querySelector(`meta[name='action-cable-${name}']`)
+    return (element ? element.getAttribute("content") : undefined)
+  },
+
+  createWebSocketURL(url) {
+    if (url && !/^wss?:/i.test(url)) {
+      const a = document.createElement("a")
+      a.href = url
+      // Fix populating Location properties in IE. Otherwise, protocol will be blank.
+      a.href = a.href
+      a.protocol = a.protocol.replace("http", "ws")
+      return a.href
+    } else {
+      return url
+    }
+  },
+
+  startDebugging() {
+    this.debugging = true
+  },
+
+  stopDebugging() {
+    this.debugging = null
+  },
+
+  log(...messages) {
+    if (this.debugging) {
+      messages.push(Date.now())
+      this.logger.log("[ActionCable]", ...messages)
+    }
+  }
+}
+
+ActionCable.Connection = Connection
+ActionCable.ConnectionMonitor = ConnectionMonitor
+ActionCable.Consumer = Consumer
+ActionCable.INTERNAL = INTERNAL
+ActionCable.Subscription = Subscription
+ActionCable.Subscriptions = Subscriptions
+
+export default ActionCable
```

Comment:
We can save a few bytes by declaring these in the export directly.

<details><summary>View diff</summary>

```diff
--- a/actioncable/app/javascript/action_cable/index.js
+++ b/actioncable/app/javascript/action_cable/index.js
@@ -5,7 +5,13 @@ import INTERNAL from "./internal"
 import Subscription from "./subscription"
 import Subscriptions from "./subscriptions"
 
-const ActionCable = {
+export default {
+  Connection,
+  ConnectionMonitor,
+  Consumer,
+  INTERNAL,
+  Subscription,
+  Subscriptions,
   WebSocket: window.WebSocket,
   logger: window.console,
 
@@ -14,7 +20,7 @@ const ActionCable = {
       const urlConfig = this.getConfig("url")
       url = (urlConfig ? urlConfig : this.INTERNAL.default_mount_path)
     }
-    return new ActionCable.Consumer(this.createWebSocketURL(url))
+    return new Consumer(this.createWebSocketURL(url))
   },
 
   getConfig(name) {
@@ -50,12 +56,3 @@ const ActionCable = {
     }
   }
 }
-
-ActionCable.Connection = Connection
-ActionCable.ConnectionMonitor = ConnectionMonitor
-ActionCable.Consumer = Consumer
-ActionCable.INTERNAL = INTERNAL
-ActionCable.Subscription = Subscription
-ActionCable.Subscriptions = Subscriptions
-
-export default ActionCable
```

</details><br>

```
app/assets/javascripts/action_cable.js
  before: 19664 bytes
   after: 19586 bytes
```

Reply:
Done in https://github.com/rmacklin/rails/commit/ad035aac03ecaed8c784d3f8bf958914cf515c6b